src/SkyWalking.Core/Remote/GrpcApplicationService.cs

```diff
@@ -45,96 +45,120 @@ public class GrpcApplicationService : TimerService
+        {
+            if (DictionaryUtil.IsNull(RemoteDownstreamConfig.Agent.ApplicationInstanceId))
+            {
+                var instanceDiscoveryService =
+                    new InstanceDiscoveryService.InstanceDiscoveryServiceClient(availableConnection.GrpcChannel);
 
-                    // todo fix Device not configured
-                    //var ipv4s = Dns.GetHostAddresses(hostName);          
-                    //foreach (var ipAddress in ipv4s.Where(x => x.AddressFamily == AddressFamily.InterNetwork))
-                    //   osInfo.Ipv4S.Add(ipAddress.ToString());
+                var agentUUID = Guid.NewGuid().ToString("N");
+                var registerTime = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
 
-                    var applicationInstance = new ApplicationInstance
-                    {
-                        ApplicationId = RemoteDownstreamConfig.Agent.ApplicationId,
-                        AgentUUID = agentUUID,
-                        RegisterTime = registerTime,
-                        Osinfo = osInfo
-                    };
+                var hostName = Dns.GetHostName();
 
-                    var applicationInstanceId = 0;
-                    var retry = 0;
+                var osInfo = new OSInfo
+                {
+                    Hostname = hostName,
+                    OsName = Environment.OSVersion.ToString(),
+                    ProcessNo = Process.GetCurrentProcess().Id
+                };
 
-                    while (retry++ <= 3 && DictionaryUtil.IsNull(applicationInstanceId))
-                    {
-                        var applicationInstanceMapping =
-                            await instanceDiscoveryService.registerInstanceAsync(applicationInstance);
-                        await Task.Delay(TimeSpan.FromSeconds(1), token);
-                        applicationInstanceId = applicationInstanceMapping.ApplicationInstanceId;
-                    }
+                // todo fix Device not configured
+                //var ipv4s = Dns.GetHostAddresses(hostName);          
+                //foreach (var ipAddress in ipv4s.Where(x => x.AddressFamily == AddressFamily.InterNetwork))
+                //   osInfo.Ipv4S.Add(ipAddress.ToString());
 
+                var applicationInstance = new ApplicationInstance
+                {
+                    ApplicationId = RemoteDownstreamConfig.Agent.ApplicationId,
+                    AgentUUID = agentUUID,
+                    RegisterTime = registerTime,
+                    Osinfo = osInfo
+                };
+
+                var retry = 0;
+                var applicationInstanceId = 0;
+                while (retry++ < 5 && DictionaryUtil.IsNull(applicationInstanceId))
+                {  
+                    var applicationInstanceMapping =await instanceDiscoveryService.registerInstanceAsync(applicationInstance);
```

Comment:
~~`registerInstanceAsync()` Whether to follow the C# code specification?~~
I find this is due to code generator code, so we can ignore that.